# Bioinformatics-in-Bacteriophages
## Leveraging the Levenshtein Distance for Enhanced Bacteriophage Genome Analysis

[Thesis Library](https://flextra.flinders.edu.au/flex/public/rhd/view/e2950f68-74f7-472a-b083-42a83656b802/1) \
Superviser: [Rob Edwards](https://github.com/linsalrob), [Susie Grigson](https://github.com/susiegriggo) and [Edward Lab](https://github.com/linsalrob/EdwardsLab)

### Abstract
The objective of this thesis is to evaluate the efficacy of genome similarity analysis by utilising the Levenshtein distance and generating phylogenetic trees. This method functions as both an alternative method for phage classification and an investigation of the extent to which the gene arrangement within genomes is consistent with the current taxonomic classification. Moreover, its efficacy is evaluated in comparison to the current classification principle. The analysis has the potential to offer valuable insights into phage classification, which could be instrumental in the comprehension of phage biology, the prediction of phage-host interactions, and the development of precise classification systems for the effective use of phages in therapy and other applications.

The phage genomes datasets were compiled from the NCBI Genbank database. Genome similarity was then computed using the Levenshtein and Mash distances. The phylogenies were constructed from the Levenshtein distance and visualised using the Interactive Tree Of Life (iTOL) online tool. Numerous phage characteristics, including genome length, bacterial host, and viral taxonomy, were employed to analyse these trees. To evaluate the correlation between the two-distance metrics, tanglegrams were generated. The potential of this method to investigate the relationship between phage gene arrangement and phage taxonomy is illustrated by the results of this study. Research in the future should expand the phage dataset and investigate in several algorithm methods.
